test/solidity/inheritance/inheritance.js

```diff
@@ -0,0 +1,54 @@
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+
+describe('Crypto Inheritance tests', function () {
+  let signers, contractMain, contractMainAddr, contractBase, contractBaseAddr, wallet
+  const TOP_UP_AMOUNT = ethers.utils.parseEther('0.000001');
+
+  before(async function () {
+    signers = await ethers.getSigners()
+    wallet = signers[0];
+
+    const factoryMain = await ethers.getContractFactory('Main')
+    contractMain = await factoryMain.deploy()
+    await contractMain.deployed()
+    contractMainAddr = contractMain.address
```

Comment:
we can remove the redundant variable `contractMainAddr` if it's used only once in `const tx = {}` and use directly `contractMain.address`

Reply:
It's used on line 34 as well